lib/mongoid/persistable.rb

```diff
@@ -165,10 +223,70 @@ def process_atomic_operations(operations)
       operations.each do |field, value|
         access = database_field_name(field)
         yield(access, value)
-        remove_change(access)
+        remove_change(access) unless executing_atomically?
       end
     end
 
+    # Remove the dirty changes for all fields changed in the current atomic
+    # context.
+    #
+    # @api private
```

Comment:
Is there any reason these can't just be private methods instead of documented as `@api private` and then prefixed with `_mongoid_`? They don't seem to be called anywhere except internally.

Reply:
These methods are already private, however since they are included in all model instances there could be confusion as to whether a model is allowed to call these methods on itself as part of our public API. The private api declaration explicitly says these methods are not part of public API.